### ISSUE TITLE
SASS AutoCompile: Error while compiling

### ISSUE THREAD

**Issue by Aubaroo** [armin-pfaeffle/sass-autocompile#12]
My error message says node-sass is not recognised as an internal or external command, operable program or batch file. 

I had no problem getting node-sass through my command line...

**Comment by Aubaroo** [comment]
The code in line 1 is 

out: ../SCSS tutorial/main.css

**Comment by armin-pfaeffle** [comment]
> I had no problem getting node-sass through my command line...

What exactly do you mena with that? Installing node-sass? Or run it via terminal?

What happens when you open terminal and enter `node-sass`?

**Comment by Aubaroo** [comment]
It says provide a Sass file to render

**Comment by Aubaroo** [comment]
I followed the instruction to install node-sass by typing npm install node-sass -g on the command prompt

**Comment by armin-pfaeffle** [comment]
What system do you use? Mac OS? Windows? Linux?

**Comment by Aubaroo** [comment]
Windows 7 64 bit

**Comment by armin-pfaeffle** [comment]
Okay... Hm, I have the same system and everything works fine here. It seems that your Atom uses another global `Path` variable. That is a lil bit strange, because you terminal finds `node-sass`.

Can you have a look at this path `C:\Users\<username>\AppData\Roaming\npm`. Is there a batch named `node-sass.cmd`?

And can you have a look at the global `Path` variable and check if the path `C:\Users\yusername>\AppData\Roaming\npm` is part of it?

**Comment by Aubaroo** [comment]
It says the C:\Users\<username>\AppData\Roaming\npm is not recognised as an internal or external command... (Yes, I did change the <username> to my actual username).

Im just started to learn code and the uses of command prompt so could you tell me how to do the second step you mentioned about global paths?

**Comment by armin-pfaeffle** [comment]
1. Check node-sass batch file:

Open your explorer and enter the path `C:\Users<username>\AppData\Roaming\npm` into the address bar. Press enter and explorer will open the directory. Then check if there is a file that is called `node-sass.bat`.

2. Checking global `Path` varialbe

Right click on Computer (icon in start menu) and select **Properties**. In the left sidebar, click on **Advanced system settings**. In the new windows select tab **Advanced**. In the lower part, there is a button called **Environment Variables**. Click on it. There are two parts now. Have a look at the upper one. Scroll down until you see the `Path` entry and make a double click on it. Then check the value if there is a path named `C:\Users\<username>\AppData\Roaming\npm`. Because then everything should be okay.

**Comment by Aubaroo** [comment]
1. There is a file named node-sass.. but Im not sure about the .bat part... The type of the file just says 'file'. 

2. Yes, its there.

**Comment by armin-pfaeffle** [comment]
Hm, that is strange... In this case, Atom should find `node-sass` command.

Do you start Atom with your normal user? Or do you use another account? Is there anything *special* about your Windows/Atom installation?

Perhaps it can help that you uninstall sass-autocompile and node-sass and reinstall both.

**Comment by Aubaroo** [comment]
Yes, the user I am using right now is the only account. 
No, I dont think there was anything special about my Windows/Atom installation.

You mean uninstall sass-autocompile from Atom? and how do you uninstall node-sass?

**Comment by armin-pfaeffle** [comment]
Yes, remove sass-autocompila via Atom and reinstall it. Perhaps something went wrong while installation.

Uninstall node-sass via terminal:
`npm uninstall node-sass -g`

Install it again:
`npm install node-sass -g`

**Comment by Aubaroo** [comment]
Aha! I did reinstall but was giving me the same error. So I closed Atom and opened it again and now it is giving a different error message. 'Invalid top-level expression'.

Does this refer to the code on my first line? which is 

out: ../SCSS tutorial/main.css

**Comment by armin-pfaeffle** [comment]
Perhaps you should remove the blank. Try a simple version:

```
// out: test.css
```

**Comment by Aubaroo** [comment]
It worked! Thank you very much! You are a life saver!

**Comment by Aubaroo** [comment]
I think the initial problem was due to me not restarting Atom when installing sass-autocompile.

**Comment by armin-pfaeffle** [comment]
But something seems to went wrong... Would be interessting what the problem was. What have you changed?

**Comment by Aubaroo** [comment]
The second problem was perhaps me not adding // before the code... because that is what I changed!

**Comment by Aubaroo** [comment]
Before: out: test.css
After : // out: test.css

and it worked.

**Comment by armin-pfaeffle** [comment]
Oh... okay... :D

Thank you! Then everything works fine ;)

**Comment by winers1290** [comment]
Just adding to this because I'm having the same problem.

Installed node-sass globally, and the PATH environment variable is correct, so is the location of the batch file inside C:\Users\<username>\AppData\Roaming\npm

I have uninstalled node-sass and reinstalled, and restarted Atom several times but having no luck.

My installation is a little unique; I am running Atom in Windows 10, but accessing the files from a shared drive from a virtual machine running Ubuntu 16.

Error log from Atom is: 'C:\Users\<username>\node_modules\node-sass\bin\node-sass' is not recognized as an internal or external command,
operable program or batch file.

Any ideas??

**Comment by armin-pfaeffle** [comment]
Sorry for my late answer. Is your problem up-to-date?!

**Comment by dominicdaviesme** [comment]
armin-pfaeffle, great job with the atom package!

I'm running Windows 10 x64 bit with atom and had this issue.

To solve it I went to your sass-autocompile docs and ensured I had latest node.js and node-sass on my system.  Once I updated node.js and installed node-sass, i restarted atom and Sass is compiling nicely.

It is compiling to the scss folder instead of the css folder but I can manually copy it over at the end of my project.

Once again thanks for the package!
